bin/tx-prover/src/commands/proxy.rs

```diff
@@ -18,6 +21,7 @@ impl StartProxy {
     ///
     /// This method will first read the config file to get the list of workers to start. It will
     /// then start a proxy with each worker as a backend.
+    #[tracing::instrument(target = MIDEN_TX_PROVER, name = "proxy:execute")]
```

Comment:
Should this get a different proxy `target`?

Reply:
Not sure, my idea was to group both services under the same target in order to be able to eventually join traces (adding the worker trace to the proxy essentially). Is that ok? Maybe there is other way.

Reply:
I'm unsure of how e.g. jaeger wants it, but I imagine one needs a common request UUID of sorts in order to aggregate a single request as it flows through.

Would the transaction ID not be good enough if we add it as a top level span in each process?

A potential issue would be someone sending the same transaction multiple times; in which case we would need to attach a more unique request identifier which is usually communicated using the `X-Request-ID` http extension header. Though this feels unnecessary at this stage, and probably we only care about this on a transaction ID level.

You already have more practical experience with jaeger though, so maybe what I'm thinking isn't applicable.

Reply:
Both from the proxy and worker perspective I think that eventually adding a `X-Request-ID` header is the best option, using the UUID created in the proxy.

Reply:
Could you create a follow-up issue for this? Likely we'll also want to do this in the node, and maybe even in the client.

Reply:
Done #1031